docs/core-concepts/route-params.md

```diff
@@ -0,0 +1,156 @@
+# Route Params
+
+When defining your route `path`, you can use a colon `:` to denote dynamic params.
+
+```ts
+import { 
+  Routes,
+} from 'kitbag/router'
+
+const routes = [
+  {
+    name: 'users',
+    path: '/users/:id',
+    component: ...
+  }
+] as const satisfies Routes
+```
+
+This means the route will expect 1+ extra string characters (anything but a forward slash `/`) in order to be considered a match. The value of these extra characters is captured and exposed in the `route.params`
+
+```ts
+const route = useRoute()
+
+route.params.id
+```
+
+When defined this way, these params are reactive strings. If you update a param, the url will be updated accordingly.
+
+## Param Types
+
+With the `path` function, kitbag/router supports parsing params to types other than `string`.
+
+```ts
+import { 
+  Routes,
+  path, // [!code ++]
+} from 'kitbag/router'
+
+const routes = [
+  {
+    name: 'users',
+    path: '/users/:id', // [!code --]
+    path: path('/users/:id', { id: Number }), // [!code ++]
+    component: ...
+  }
+] as const satisfies Routes
+```
+
+This will automatically parse the param from `string` in the URL to `number` in `route.params`. If the value cannot be parsed, the route will not be considered a match.
+
+kitbag/router ships with support for `String` (default), `Boolean`, `Number`, and `RegExp`.
+
+### RegExp Params
+
+Using native RegExp is a powerful way of controlling route matching.
+
+```ts
+const routes = [
+  {
+    name: 'users',
+    path: path('/users/:id', { id: /^A[0-9]$/  }), // [!code focus]
+    component: ...
+  }
+] as const satisfies Routes
+```
+
+### Custom Param
+
+You're not limited to the param types that ship with kitbag/router, use `ParamGetter<T>` or `ParamGetSet<T>` to parse params to whatever type you need.
+
+```ts
+type MyType = {
+  version: number,
+  subversion?: string,
+}
+
+const myTypeParam: ParamGetter<MyType> = (value, { invalid }) => {
+  // expected input format '123.alpha'
+  const [versionString, subversion] = value.split('.')
+  const version = parseInt(versionString)
+
+  if (isNaN(version)) {
+    // If any exception is thrown, the route will not match.
+    // Use the provided `invalid` function to provide additional context to the router.
+    throw invalid('Value provided for version is not valid integer')
+  }
+
+  // Return value is what will be provided in route.params.id
+  return { version, subversion }
+}
+```
+
+Update your param assignment on the route's path
+
+```ts
+const routes = [
+  {
+    name: 'users',
+    path: path('/users/:id', { id: myTypeParam }), // [!code focus]
+    component: ...
+  }
+] as const satisfies Routes
+```
+
+With this getter defined, now our route will only match if the param matches our rules above.
+
+As a `ParamGetter`, the value in `route.params` is still writable, but the set will assume `value.toString()` is sufficient. Alternatively if you use `ParamGetSet`, you can provide the same validation on value set as well.
```

Comment:
Since `ParamGetter` expects to be able to call `toString` we should probably have this example work with that. And maybe use this example for `ParamGetSet`

Reply:
yep, great call